digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/AddressLinkBuilder.scala

```diff
@@ -213,7 +214,7 @@ trait AddressLinkBuilder {
           right.map(_.copy(segmentMValue = if (nextSegment.sideCode == SideCode.AgainstDigitizing) startMValue else endMValue)))
       }
 
-      if(nextSegment.sideCode.value != previousSegment.sideCode.value)
+      if(nextSegment.sideCode.value != previousSegment.sideCode.value && GeometryUtils.geometryLength(previousSegment.geometry) != 0 && GeometryUtils.geometryLength(nextSegment.geometry) != 0)
```

Comment:
should we change the message to log something about geometry length being zero?

Reply:
Don't think so. This is just to prevent the cases that we have only one point in geometry, and then it gets random sidecode value, since it's impossible to build a direction of vector with just one point